pkg/datapath/linux/routing/migrate.go

```diff
@@ -0,0 +1,628 @@
+				scopedLog.WithError(err).
+					Error("Failed to migrate endpoint to new ENI datapath due to failure in listing routes associated with rule. " +
+						"Previous datapath is still intact and endpoint connectivity is not affected.")
+
+				failed++
+				continue
+			}
+
+			// If there are no routes under the same table as the rule, then
+			// skip.
+			if len(routes) == 0 {
+				scopedLog.Debug("Skipping migration of egress rule due to no routes found, proceeding to delete")
+
+				// Best-effort cleanup. It's not worth stopping the migration
+				// and start reverting for a rule that has no direct impact on
+				// the datapath. If there's a collision in the future with this
+				// rule, then it'll fall-through to the next match in the RPDB
+				// because it has no routes.
+				// if _, err := m.deleteOldRule(r); err != nil {
+				// 	scopedLog.WithError(err).Warn("Failed to delete rule that had no routes associated with it")
+				// }
+
+				continue
+			}
+
+			// It is sufficient to grab the first route that matches because we
+			// are assuming all routes created under a rule will have the same
+			// ifindex (LinkIndex).
+			ifindex := routes[0].LinkIndex
+			newTable, err := m.retrieveTableIDFromIfIndex(ifindex)
+			if err != nil {
+				scopedLog.WithError(err).
+					Error("Failed to migrate endpoint to new ENI datapath due to failure in retrieving new table ID. " +
+						"Previous datapath is still intact and endpoint connectivity is not affected.")
+
+				failed++
+				continue
+			}
+
+			var (
+				stack revert.RevertStack
+
+				oldTable = r.Table
+			)
+
+			revert, err := m.copyRoutes(routes, oldTable, newTable)
+			stack.Extend(revert)
+			if err != nil {
+				scopedLog.WithError(err).
+					Error("Failed to migrate endpoint to new ENI datapath due to failure in creating new routes. " +
+						"Previous datapath is still intact and endpoint connectivity is not affected.")
+
+				failed++
+				continue
+			}
+
+			revert, err = m.createNewRule(
+				r,
+				linux_defaults.RulePriorityEgress,
+				oldTable,
+				linux_defaults.RulePriorityEgressv2,
+				newTable,
+			)
+			stack.Extend(revert)
+			if err != nil {
+				// We revert here because we want to ensure that the new routes
+				// are removed as they'd have no effect, but may conflict with
+				// others in the future.
+				if revErr := stack.Revert(); revErr != nil {
+					msg := "Failed to migrate endpoint to new ENI datapath and failed to cleanup new rule and routes. " +
+						"Previous datapath is still intact and will continue to operate for the endpoint. " +
+						"It is advised to investigate these errors."
+					scopedLog.WithError(err).WithField("revertError", revErr).Error(msg)
+				} else {
+					scopedLog.WithError(err).
+						Info("Failed to migrate endpoint to new ENI datapath, but was successfully reverted. " +
+							"Previous datapath is still intact and will continue to operate for the endpoint.")
+				}
+
+				failed++
+				continue
```

Comment:
With a dedicated function for the inner logic, these lines would boil down to:

```
return fmt.Errorf("Failed to insert new routing rule")
```

Which is clear, concise, and easier to follow than trying to scan through and skip past the multiline logging logic handling.